src/libraries/Microsoft.PowerFx.Connectors/Environment/CdpTabularService.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    // Implements CDP protocol for Tabular connectors
+    public class CdpTabularService : TabularService
+    {
+        public string DataSetName { get; }
+
+        public string TableName { get; }
+
+        protected HttpClient _httpClient;
```

Comment:
>protected HttpClient _httpClient; [](http://example.com/codeflow?start=8&length=33)

Is it safe to hold this as a field? 
We may need to get from IServiceProvider  (which is why we pass SP ) 

Maybe:
virtual HttpClient GetClient(IServiceProvider)

Reply:
instead of passing an HttpClient in the constructor, I'll use Func<HttpClient> like for the jwt token in PowerAppsClient